README.md

```diff
@@ -1,53 +1,78 @@
-# Open Source Project Template
+# HMAC Signed requests in dart
 
-This repository contains a template to seed a repository for an Open Source
-project.
+This Dart package provides a convenient way to make HTTP requests with HMAC authentication. HMAC (Hash-based Message Authentication Code) authentication is a mechanism for verifying the authenticity of a message using a cryptographic hash function and a secret key.
 
-## How to use this template
+## Features
 
-1. Check out this repository
-2. Delete the `.git` folder
-3. Git init this repository and start working on your project!
-4. Prior to submitting your request for publication, make sure to review the
-   [Open Source guidelines for publications](https://nventive.visualstudio.com/Internal/_wiki/wikis/Internal_wiki?wikiVersion=GBwikiMaster&pagePath=%2FOpen%20Source%2FPublishing&pageId=7120).
+- Supports HMAC authentication for secure requests in HTTP and HTTPS.
+- Core Encryption Algorithm is SHA-256.
+- Provides classes for GET, POST, PUT, and DELETE requests with HMAC authentication.
+- Easily configurable and customizable for different APIs.
+- Depends on [crypto](https://pub.dev/packages/crypto) and [http](https://pub.dev/packages/http).
 
-## Features (to keep as-is, configure or remove)
-- [Mergify](https://mergify.io/) is configured. You can edit or remove [.mergify.yml](/.mergify.yml).
+## Installation
 
-The following is the template for the final README.md file:
+### Use this package as a library.
 
----
+Depend on it
 
-# Project Title
+##### Run this command:
 
-{Project tag line}
+```
+flutter pub add requests_signature_dart
+```
 
-{Small description of the purpose of the project}
+This will add a line like this to your package's `pubspec.yaml` (and run an implicit `flutter pub get`):
 
-[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE)
+```
+dependencies:
+  request_signature_dart: ^1.0.0
+```
+Alternatively, your editor might support `flutter pub get`. Check the docs for your editor to learn more.
 
-## Getting Started
+##### Import it 
 
-{Instructions to quickly get started using the project: pre-requisites, packages
-to install, sample code, etc.}
+in your dart code:
 
-## Features
+```
+import 'package:requests_signature_dart/request_signature_dart.dart';
+```
+
+## Usage
+
+### In your dart code
+
+```
+import 'package:requests_signature_dart/requests_signature_dart.dart';
 
-{More details/listing of features of the project}
+void main() async {
+  // Initialize SignedHMACRequest with your API key and secret
+  final signedRequest = SignedHMACRequest(apiKey, apiSecret);
```

Comment:
The usage described here seem too different from the original's (https://github.com/nventive/RequestsSignature?tab=readme-ov-file#implementing-requests-signature-in-http-client). If you don't want to deal with the readme in this PR, just undo the changes in it. Otherwise, the readme needs to follow the guidelines and the dart recipe must align with the original .NET recipe.